### ISSUE TITLE
Add more descriptive exceptions

### ISSUE THREAD

**Issue by kfichter** [omisego/plasma-mvp#57]
We're currently handling a significant portion of our exceptions as `AssertionError`. This is bad practice because it means we end up with unit tests that look like this:

https://github.com/omisego/plasma-mvp/blob/cf479aa6a9ec5baf64b0ca64db521f657dfa78a4/tests/child_chain/test_child_chain.py#L45-L60

There's absolutely no way to determine if the `AssertionError` being caught in test#1 is actually the correct `AssertionError`. We should create better and more descriptive exceptions. 

There's a few ways to do this properly:
1. Continue to use `AssertionError`, but attach messages to the `assert` statement and verify those messages (samples thanks to @boolafish):

Modify the assert:
```
assert valid_signature, 'invalid signature'
```

Modify the test:
```
with pytest.raises(AssertionError) as e:
    child_chain.submit_block(block)
assert str(e.value) == 'invalid signature'
```

2. Extend `Exception` with more descriptive errors, e.g.:
```
class InvalidBlockSignatureError(Exception):
    """Raise when a block is signed by anyone but the operator"""
```

I'm personally a fan of (2), but I couldn't find any sort of guide on which model to generally stick to.

**Comment by boolafish** [comment]
as for reference: https://stackoverflow.com/questions/20076182/what-are-the-implications-of-using-assert-vs-raise-exception

**Comment by kfichter** [comment]
Fixed in #62
